cpp/include/cudf/column/column.hpp

```diff
@@ -75,6 +76,29 @@ class column {
    */
   column(column&& other) noexcept;
 
+  /**
+   * @brief Construct a new column by taking ownership of the contents of a device_uvector.
+   *
+   * @param other The device_uvector whose contents will be moved into the new column.
+   * @param null_mask Optional, column's null value indicator bitmask. May
+   * be empty if `null_count` is 0 or `UNKNOWN_NULL_COUNT`.
+   * @param null_count Optional, the count of null elements. If unknown, specify
+   * `UNKNOWN_NULL_COUNT` to indicate that the null count should be computed on
+   * the first invocation of `null_count()`.
+   */
+  template <typename T, CUDF_ENABLE_IF(cudf::is_numeric<T>() or cudf::is_chrono<T>())>
+  column(rmm::device_uvector<T>&& other,
+         rmm::device_buffer&& null_mask = {},
+         size_type null_count           = UNKNOWN_NULL_COUNT)
+  {
+    CUDF_EXPECTS(other.size() <= std::numeric_limits<size_type>::max());
```

Comment:
Let's add `static_cast` to avoid any potential compiler warning.
```suggestion
    CUDF_EXPECTS(other.size() <= static_cast<std::size_t>(std::numeric_limits<size_type>::max()));
```

Reply:
Just use `other.ssize()`

https://github.com/rapidsai/rmm/blob/be9b9a96aecac44ffb03ee9681ce1cf17027be65/include/rmm/device_uvector.hpp#L490

Reply:
That seems dangerous too: https://github.com/rapidsai/rmm/blob/be9b9a96aecac44ffb03ee9681ce1cf17027be65/include/rmm/device_uvector.hpp#L490--L494

If `other.size()` exceeds `std::numeric_limits<int64_t>::max()`, the returned value will also overflow.

Reply:
I don't think the signed size is quite what we're looking for because `int64_t` is still not the same as `size_type`. We're picking between implicit upcasts and working around unsigned/signed comparisons, and I'd rather do the latter. The most straightforward path to me seems to be Nghia's suggestion.